lib/output.js

```diff
@@ -1041,18 +1041,18 @@ function OutputStream(options) {
         self._do_print_body(output);
     });
     DEFPRINT(AST_ForIn, function(self, output){
+        var is_for_of = self instanceof AST_ForOf;
+        var is_for_of_sequence = is_for_of && self.object instanceof AST_Sequence;
         output.print("for");
         output.space();
         output.with_parens(function(){
             self.init.print(output);
             output.space();
-            if (self instanceof AST_ForOf) {
-                output.print("of");
-            } else {
-                output.print("in");
-            }
+            output.print(is_for_of ? "of" : "in");
             output.space();
+            if (is_for_of_sequence) output.print("(");
             self.object.print(output);
+            if (is_for_of_sequence) output.print(")");
```

Comment:
Wouldn't it be easier to add `p instanceof AST_ForOf` to
https://github.com/mishoo/UglifyJS2/blob/0f2be1456c42d470283c7e98b27e10c6a83d5b16/lib/output.js#L755-L770

Reply:
It's less efficient and more error prone in my opinion. I could put it there if you want.